warp/backend.go

```diff
@@ -88,9 +85,8 @@ func NewBackend(
 		warpSigner:                warpSigner,
 		blockClient:               blockClient,
 		signatureCache:            signatureCache,
-		uptimeCalculator:          uptimeCalculator,
-		validatorState:            validatorsState,
-		stateLock:                 stateLock,
+		validatorReader:           validatorReader,
+		validatorLock:             validatorLock,
```

Comment:
Seem a bit odd that we pass this interface but also a lock for it separately?
maybe some kind of locker wrapper for `GetValidator` / `CalculateUptime` is appropriate here? does the lock need to be held atomically for both operations?
This can be out of scope for this PR.

Reply:
yes they should be locked together. I added another method to interface that locks the chainLock.